server/src/controllers/course.ts

```diff
@@ -93,19 +118,116 @@ function parseSisuInstance(instance: SisuInstance): InstanceData {
   };
 }
 
+// Middleware function for handling errors in JSON parsing
+export function handleInvalidRequestJson(err: JsonError, req: Request, res: Response, next: NextFunction): void {
+  if (err instanceof SyntaxError && err.status === 400 && err.body !== undefined) {
+    res.status(400).send({
+      success: false,
+      errors: [ `SyntaxError: ${err.message}: ${err.body}` ]
+    });
+  } else
+    next();
+}
 
 export async function addCourse(req: Request, res: Response): Promise<void> {
+  const requestSchema: yup.AnyObjectSchema = yup.object().shape({
+    courseCode: yup.string().required(),
+    department: localizedStringSchema.required(),
+    name: localizedStringSchema.required()
+  });
+
+  const t: Transaction = await sequelize.transaction();
+
   try {
-    // TODO: add the course to the database
-    res.send({
-      success: true
+    await requestSchema.validate(req.body, { abortEarly: false });
+
+    const course: Course = await Course.create({
+      courseCode: req.body.courseCode
+    }, { transaction: t });
+
+    const courseTranslations: CourseTranslation[] = await CourseTranslation.bulkCreate([
```

Comment:
Unless I'm forgetting something, we've used `Array<T>` everywhere else for array type annotations, we should be consistent. This could also be enforced by the linter.

Reply:
+1 for enforcing one style with linter

Reply:
Now enforced by eslint